cabal-install/Distribution/Client/CmdClean.hs

```diff
@@ -0,0 +1,125 @@
+-- | cabal-install CLI command: clean
+--
+module Distribution.Client.CmdClean (
+    cleanCommand,
+    cleanAction,
+  ) where
+
+import Distribution.Client.DistDirLayout
+         ( DistDirLayout(..), defaultDistDirLayout )
+import Distribution.Client.ProjectConfig
+         ( findProjectRoot )
+import qualified Distribution.Client.Setup as Client
+import Distribution.Client.Setup
+         ( GlobalFlags )
+import Distribution.PackageDescription
+         ( emptyHookedBuildInfo, updatePackageDescription )
+import Distribution.PackageDescription.Configuration
+         ( flattenPackageDescription )
+import Distribution.PackageDescription.Parse
+         ( readPackageDescription )
+import Distribution.Simple.Command
+         ( CommandUI(..), noExtraFlags, option, usageAlternatives )
+import Distribution.Simple.Configure
+         ( findDistPrefOrDefault, maybeGetPersistBuildConfig
+         , writePersistBuildConfig )
+import Distribution.Simple.Setup
+         ( CleanFlags(..), defaultCleanFlags, defaultDistPref, fromFlag
+         , fromFlagOrDefault, optionDistPref, optionVerbosity, toFlag, trueArg )
+import Distribution.Simple.Utils
+         ( chattyTry, defaultPackageDesc, notice )
+import Distribution.Types.PackageDescription
+         ( PackageDescription(..) )
+
+import Control.Monad (when)
+import Data.Foldable (forM_, traverse_)
+import System.Directory
+import System.FilePath ((</>))
+
+cleanCommand :: CommandUI CleanFlags
+cleanCommand = Client.installCommand {
+  commandName         = "new-clean",
+  commandSynopsis     = "Cleans up after a Nix-local build.",
+  commandUsage        = usageAlternatives "new-clean" [ "[FLAGS]" ],
+  commandDescription  = Just $ \_ ->
+    "Cleans up intermediate build files such as .hi, .o, preprocessed sources, etc.",
+  commandDefaultFlags = defaultCleanFlags,
+  commandNotes        = Just $ \pname ->
+        "Examples:\n"
+     ++ "  " ++ pname ++ " new-clean           "
+     ++ "    Clean the package in the current directory\n",
+  commandOptions      = \showOrParseArgs ->
+    [optionVerbosity cleanVerbosity (\v flags -> flags { cleanVerbosity = v })
+    ,optionDistPref
+       cleanDistPref (\d flags -> flags { cleanDistPref = d })
+       showOrParseArgs
+    ,option "s" ["save-configure"]
+       "Do not remove the configuration file (dist/setup-config) during cleaning. Saves need to reconfigure."
+       cleanSaveConf (\v flags -> flags { cleanSaveConf = v })
+       trueArg
+    ]
+  }
+
+cleanAction :: CleanFlags -> [String] -> GlobalFlags -> IO ()
+cleanAction flags args _ = do
+    distPref <- findDistPrefOrDefault (cleanDistPref flags)
+    let flags' = flags { cleanDistPref = toFlag distPref }
+    noExtraFlags args
+
+    pdfile <- defaultPackageDesc verbosity
```

Comment:
This looks wrong. There's no reason to believe you actually have a specific package if you new-clean. Say you're in the top-level directory of a project. What is `pdfile` going to be here? Bogus.

Meta-commentary, I probably would NOT go about implementing this as a copy paste from the existing clean. Better to just do it from scratch, I think.